src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,42 +1,76 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
+import { setCurrentTodo } from '../../features/currentTodo';
 
 export const TodoModal: React.FC = () => {
+  const currentTodo = useAppSelector(state => state.currentTodo);
+  const [currentUser, setCurrentUser] = useState<User>();
+  const [isLoading, setIsLoading] = useState(false);
+  const dispatch = useAppDispatch();
+
+  useEffect(() => {
+    setIsLoading(true);
+    if (!currentTodo) {
+      setIsLoading(false);
+
+      return;
+    }
+
+    getUser(currentTodo.id).then(user => {
+      setCurrentUser(user);
+      setIsLoading(false);
+    });
```

Comment:
Consider adding error handling for the `getUser` promise. If the API call fails, the `isLoading` state will remain true, and the user will not be informed of the error.